lib/ttfunk/table/cff/dict.rb

```diff
@@ -0,0 +1,225 @@
+require 'bigdecimal'
+
+module TTFunk
+  class Table
+    class Cff < TTFunk::Table
+      class Dict < TTFunk::SubTable
+        include Enumerable
+
+        def [](operator)
+          @dict[operator]
+        end
+
+        def each(&block)
+          @dict.each(&block)
+        end
+
+        alias each_pair each
+
+        def encode
+          ''.tap do |result|
+            each do |(operator, operands)|
+              operands.each { |operand| result << encode_operand(operand) }
+              result << encode_operator(operator)
+            end
+          end
+        end
+
+        private
+
+        def encode_operator(operator)
+          if operator >= 1200
+            [12, operator - 1200].pack('C*')
+          else
+            [operator].pack('C')
+          end
+        end
+
+        def encode_operand(operand)
+          bytes = case operand
+                  when Integer
+                    encode_integer(operand)
+                  when Float, BigDecimal
+                    encode_float(operand)
+                  when Sci
+                    encode_sci(operand)
+                  end
+
+          bytes.pack('C*')
+        end
+
+        def encode_integer(int)
+          case int
+          when -107..107
+            [int + 139]
+
+          when 108..1131
+            int -= 108
+            [(int >> 8) + 247, int & 0xFF]
+
+          when -1131..-108
+            int = -int - 108
+            [(int >> 8) + 251, int & 0xFF]
+
+          when -32_768..32_767
+            [28, (int >> 8) & 0xFF, int & 0xFF]
+
+          else
+            encode_integer32(int)
+          end
+        end
+
+        def encode_integer32(int)
+          [
+            29,
+            (int >> 24) & 0xFF,
+            (int >> 16) & 0xFF,
+            (int >> 8) & 0xFF,
+            int & 0xFF
+          ]
```

Comment:
Yeah, this is great. Didn't know about `V`.

> Also maybe all `encode_` methods should return string instead of byte arrays to facilitate faster encoding methods than manually unpacked integers.

This is already the case for everything except `encode_integer` no?